oxli/khmer_api.py

```diff
@@ -0,0 +1,196 @@
+        read1.sequence = read1.sequence.replace('N', 'A')
+
+        if read2:
+            read2.sequence = read2.sequence.upper()
+            read2.sequence = read2.sequence.replace('N', 'A')
+
+        yield n, is_pair, read1, read2
+
+
+def output_reads(input_stream, out_fp):
+    for n, is_pair, read1, read2 in input_stream:
+        if is_pair:
+            write_record_pair(read1, read2, out_fp)
+        else:
+            write_record(read1, out_fp)
+
+
+def build_graph(ifilenames, graph, num_threads = 1, tags = False):
+    
+    if(tags):
+        eat = graph.consume_fasta_and_tag_with_reads_parser
+    else:
+        eat = graph.consume_fasta_with_reads_parser
+    
+    for _, ifile in enumerate(ifilenames):
+        rparser = khmer.ReadParser(ifile)
+        threads = []
+    
+        for num in xrange(num_threads):
+            cur_thread = threading.Thread( target=eat, args=(rparser,))
+            threads.append(cur_thread)
+            cur_thread.start()
+
+        for thread in threads:
+            thread.join()
+        
+
+def diginorm(input_stream, ct, coverage, ksize=0, ofp = False):
+    n = 0
+    discard = 0
+    passed = False
+    for _, is_pair, read1, read2 in input_stream:
+        if is_pair:
+            med1, _, _ = ct.get_median_count(read1.sequence)
+            med2, _, _ = ct.get_median_count(read2.sequence)
+
+            if (med1 < coverage and len(read1.sequence) > ksize) \
+                    or (med2 < coverage and len(read2.sequence) > ksize):
+                ct.consume(read1.sequence)
+                ct.consume(read2.sequence)
+                yield n, True, read1, read2
+                n += 2
+            else:
+                discard += 2
+                passed = False
+        else:
+            med, _, _ = ct.get_median_count(read1.sequence)
+            if med < coverage and len(read1.sequence) > ksize:
+                ct.consume(read1.sequence)
+                yield n, False, read1, None
+                n += 1
+            else:
+                discard += 1
+
+
+def trim(input_stream, ct, normalize_coverage, trusted_coverage):
+    n = 0
+    for _, is_pair, read1, read2 in input_stream:
+        if is_pair:
+            med1, _, _ = ct.get_median_count(read1.sequence)
+            med2, _, _ = ct.get_median_count(read2.sequence)
+
+            if med1 >= normalize_coverage and med2 >= normalize_coverage:
+                read1 = _trim_record(read1, ct, trusted_coverage)
+                read2 = _trim_record(read2, ct, trusted_coverage)
+            yield n, True, read1, read2
+            n += 2
+        else:
+            med, _, _ = ct.get_median_count(read1.sequence)
+            if med >= normalize_coverage:
+                read1 = _trim_record(read1, ct, trusted_coverage)
+            yield n, False, read1, None
+            n += 1
+
+
+def streamtrim(input_stream, ct, normalize_coverage, trusted_coverage):
+    import khmer.utils
```

Comment:
Why not import `broken_paired_reader` on line https://github.com/ged-lab/khmer/pull/961/files#diff-0fe57386cad3c94ff54bb7daece9b655R12 ?